src/qt/pivx/settings/settingswalletoptionswidget.cpp

```diff
@@ -60,17 +60,14 @@ SettingsWalletOptionsWidget::SettingsWalletOptionsWidget(PIVXGUI* _window, QWidg
     connect(ui->pushButtonSave, &QPushButton::clicked, [this] { Q_EMIT saveSettings(); });
     connect(ui->pushButtonReset, &QPushButton::clicked, this, &SettingsWalletOptionsWidget::onResetClicked);
     connect(ui->pushButtonClean, &QPushButton::clicked, [this] { Q_EMIT discardSettings(); });
-
-    connect(ui->pushButtonSave, &QPushButton::clicked, [this](){
-        ClientModel::mapPort(ui->mapPortUpnp->isChecked(), ui->mapPortNatpmp->isChecked());
-    });
 }
 
 void SettingsWalletOptionsWidget::onResetClicked()
 {
     QSettings settings;
     walletModel->resetWalletOptions(settings);
     clientModel->getOptionsModel()->setNetworkDefaultOptions(settings, true);
+    saveMapPortOptions();
```

Comment:
This has the opposite effect, inside `saveMapPortOptions` you are obtaining the value from the checkboxes. So example: the user had upnp disabled, checks the checkbox, then press the "reset to default" button --> this will trigger the `mapPort()` function with the upnp flag in true, enabling upnp support when the user just wanted to reset the value to default.

Reply:
the status of the two checkboxes is reset prior to this call (the line above). without this the visual contents of the checkboxes are reset to their defaults but the data is never relayed to the appropriate function in mapport.

Reply:
hmm, just tested it again and it works fine now. In the previous branch this wasn't working.

Reply:
hmm, just tested it again and it works fine now. In the previous branch this wasn't working.